ozaria/site/components/play/PageIntroLevel/index.vue

```diff
@@ -69,163 +61,137 @@
           try {
-            const currentLevel = this.campaignData.levels[this.introLevelData.original]
-            const nextLevelData = getNextLevelForLevel(currentLevel) || {}
-            const nextLevelOriginal = nextLevelData.original
-            this.nextLevelStage = nextLevelData.nextLevelStage
-            if (nextLevelOriginal) {
-              this.nextLevel = await api.levels.getByOriginal(nextLevelOriginal)
-            }
+            this.introLevelSession.state.complete = true
+            await api.levelSessions.update(this.introLevelSession)
           } catch (err) {
-            console.error('Error in fetching next level', err)
+            console.error('Error in saving intro level session', err)
             // TODO handle_error_ozaria
-            noty({ text: 'Error in fetching next level', type: 'error', timeout: 2000 })
-          }
-        },
-        fetchNextLevelLink: function () {
-          if (this.nextLevel.slug && this.nextLevel.type) {
-            const nextLevelOptions = {
-              courseId: this.courseId,
-              courseInstanceId: this.courseInstanceId,
-              campaignId: this.campaignId,
-              codeLanguage: this.codeLanguage,
-              nextLevelStage: this.nextLevelStage
-            }
-            return getNextLevelLink(this.nextLevel, nextLevelOptions)
-          } else {
-            console.log('no next level found')
-            return `/ozaria/play/${this.campaignData.slug}`
+            return noty({ text: 'Error in saving intro level session', type: 'error', timeout: 2000 })
           }
         }
       }
-    )
+    }
   })
 </script>
 
 <template>
   <div v-if="dataLoaded">
     <interactives-component
       v-if="currentContent.type == 'interactive'"
-      :key="currentContentId"
+      :key="currentContentId + `${reloadKey[currentContent.type]}`"
       :interactive-id-or-slug="currentContentId"
       :code-language="language"
       @completed="onContentCompleted"
     />
     <cinematics-component
       v-else-if="currentContent.type == 'cinematic'"
-      :key="currentContentId"
+      :key="currentContentId + `${reloadKey[currentContent.type]}`"
       :cinematic-id-or-slug="currentContentId"
       :user-options="{ programmingLanguage: language }"
       @completed="onContentCompleted"
     />
     <cutscene-video-component
       v-else-if="currentContent.type == 'cutscene-video'"
-      :key="currentContentId"
+      :key="currentContentId + `${reloadKey[currentContent.type]}`"
       :cutscene-id="currentContentId"
       @completed="onContentCompleted"
     />
     <avatar-selection-screen
       v-else-if="currentContent.type == 'avatarSelectionScreen'"
-      v-on:completed="onContentCompleted">
-    </avatar-selection-screen>
+      @completed="onContentCompleted"
+    />
+    <modal-victory
+      v-if="showVictoryModal"
+      :campaign-handle="introLevelData.campaign"
```

Comment:
Could campaign specifically be driven from Vuex?

Reply:
We just need the campaign id/slug that the level is part of(which is available in the level data and thus vuex not required), and later the complete data for the campaign is driven from vuex in the victory modal.

Reply:
Could it be useful to store the current level in Vuex for other systems that need to know this info? Not priority.

Reply:
Yeah it would definitely be useful to move levels to shared Vuex store at some point.